### PR TITLE
Envelope for sparsity of clarkson transform

### PR DESCRIPTION
**Pull request by kdhingra307**
This is in conjunction with the issue #8498

Purpose of this PR is to add support for sparsity of Clarkson transformation.

### CODE REVIEW

File: scipy/linalg/_sketches.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 import numpy as np
 
 from scipy._lib._util import check_random_state
+import scipy.sparse
```
Comment (ilayn) [inline scipy/linalg/_sketches.py]: you can instead use `from scipy.sparse import csr_matrix` to keep things a bit more concise.

File: scipy/linalg/_sketches.py [hunk 2]
```diff
@@ -43,12 +44,12 @@ def cwt_matrix(n_rows, n_columns, seed=None):
...
     Where epsilon is related to the size of S
     """
-    S = np.zeros((n_rows, n_columns))
+    S = scipy.sparse.csr_matrix((n_rows, n_columns))
     nz_positions = np.random.randint(0, n_rows, n_columns)
     rng = check_random_state(seed)
     values = rng.choice([1, -1], n_columns)
     for i in range(n_columns):
-        S[nz_positions[i]][i] = values[i]
+        S[nz_positions[i],i] = values[i]
```
Comment (ilayn) [inline scipy/linalg/_sketches.py]: PEP8 one space after comma

Comment (kdhingra307) [reply scipy/linalg/_sketches.py]: @ilayn done that, my bad

File: scipy/linalg/_sketches.py (hunk 1 above)
Comment (kdhingra307) [reply scipy/linalg/_sketches.py]: thanks, did that

**Comment by rgommers** [comment]
This PR remained pretty empty (there's one line changed from `ndarray` to `csr_matrix` only) and is now superseded by gh-9884 which fully implements sparsity support. So closing this. Thanks for contributing @kdhingra307

**Pull request closed without merging**